src/Repository/ProductCommentRepository.php

```diff
@@ -166,6 +166,45 @@ public function getAverageGrade($productId, $validatedOnly)
         return (float) $qb->execute()->fetchColumn();
     }
 
+    /**
+     * @param array $productIds
+     * @param bool $validatedOnly
+     *
+     * @return array
+     */
+    public function getAverageGrades(array $productIds, $validatedOnly)
+    {
+        $sql = 'SELECT';
+
+        $count = count($productIds);
+
+        foreach ($productIds as $index => $id) {
+            $esqID = pSQL($id);
+
+            $sql .= ' SUM(IF(id_product = ' . $esqID . ' AND deleted = 0';
+            if ($validatedOnly) {
+                $sql .= ' AND validate = 1';
+            }
+            $sql .= ' ,grade, 0))';
+            $sql .= ' / SUM(IF(id_product = ' . $esqID . ' AND deleted = 0';
+            if ($validatedOnly) {
+                $sql .= ' AND validate = ';
```

Comment:
@Oksydan @PierreRambaud there's a little bug here, you should have `validate = 1` here